Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -1004,50 +1015,52 @@ u64 DirectoryBlobPartition::SetApploader(std::vector<u8> apploader, const std::s
 {
   bool success = false;
 
-  m_apploader = std::move(apploader);
-  if (m_apploader.size() < 0x20)
+  if (apploader.size() < 0x20)
   {
     ERROR_LOG_FMT(DISCIO, "{} couldn't be accessed or is too small", log_path);
   }
   else
   {
-    const size_t apploader_size = 0x20 + Common::swap32(*(u32*)&m_apploader[0x14]) +
-                                  Common::swap32(*(u32*)&m_apploader[0x18]);
-    if (apploader_size != m_apploader.size())
+    const size_t apploader_size =
+        0x20 + Common::swap32(*(u32*)&apploader[0x14]) + Common::swap32(*(u32*)&apploader[0x18]);
+    if (apploader_size != apploader.size())
       ERROR_LOG_FMT(DISCIO, "{} is the wrong size... Is it really an apploader?", log_path);
     else
       success = true;
   }
 
   if (!success)
   {
-    m_apploader.resize(0x20);
+    apploader.resize(0x20);
     // Make sure BS2 HLE doesn't try to run the apploader
-    Write32(static_cast<u32>(-1), 0x10, &m_apploader);
+    Write32(static_cast<u32>(-1), 0x10, &apploader);
   }
 
-  m_contents.AddReference(APPLOADER_ADDRESS, m_apploader);
+  size_t apploader_size = apploader.size();
+  m_contents.Add(APPLOADER_ADDRESS, std::move(apploader));
 
   // Return DOL address, 32 byte aligned (plus 32 byte padding)
-  return Common::AlignUp(APPLOADER_ADDRESS + m_apploader.size() + 0x20, 0x20ull);
+  return Common::AlignUp(APPLOADER_ADDRESS + apploader_size + 0x20, 0x20ull);
 }
 
-u64 DirectoryBlobPartition::SetDOLFromFile(const std::string& path, u64 dol_address)
+u64 DirectoryBlobPartition::SetDOLFromFile(const std::string& path, u64 dol_address,
+                                           std::vector<u8>& disc_header)
```

Comment:
Since `disc_header` is being modified, it should be a pointer as per our coding style.

```suggestion
                                           std::vector<u8>* disc_header)
```